From a93b531f8683fd09ae8e4203ba2db1003035058b Mon Sep 17 00:00:00 2001
From: Wei Qin <qinwei61@huawei.com>
Date: Tue, 18 Aug 2026 10:48:43 +0800
Subject: [PATCH 1/1] udma: bugfix related to dtu_en and sq_reserved

driver inclusion
category: bugfix
bugzilla: https://atomgit.com/openeuler/umdk/issues/842
CVE: NA

-----------------------------------------------------------

This patch fix a bug related to dtu_en and sq_reserverd.

Fixes: a7ea7e683ec6 ("udma: support alloc and free tid")

Signed-off-by: Wei Qin <qinwei61@huawei.com>
---
 src/urma/hw/udma/udma_u_jfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/urma/hw/udma/udma_u_jfs.c b/src/urma/hw/udma/udma_u_jfs.c
index 00fffff..e5eb0cc 100644
--- a/src/urma/hw/udma/udma_u_jfs.c
+++ b/src/urma/hw/udma/udma_u_jfs.c
@@ -94,7 +94,7 @@ static bool udma_check_atomic_len(uint32_t len, uint8_t opcode)
 int udma_u_set_sq_by_resp(struct udma_u_jetty_queue *sq,
 			  struct udma_create_jetty_resp *resp)
 {
-	if (!sq->dtu_en && (!sq->sq_reserved || sq->cstm))
+	if (sq->cstm || (!sq->dtu_en && !sq->sq_reserved))
 		return 0;
 
 	if (resp->buf_addr == 0) {
-- 
2.28.0.windows.1

